.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,88 @@
+name: Pipeline [test -> build -> deploy]
+
+on:
+  push:
+    branches:
+      - main
+      # - '**'
+
+  workflow_dispatch:
+    inputs:
```

Comment:
Wonder if on a workflow dispatch we need all these options?

Reply:
I've reviewed the process for CircleCI and it carries out the workflow for **all** branches, but only builds and deploys on main. I've updated this PR to reflect that.